internal/grpcwrap/provider6.go

```diff
@@ -891,6 +918,219 @@ func (p *provider6) ConfigureStateStore(ctx context.Context, req *tfplugin6.Conf
 	panic("not implemented")
 }
 
+func (p *provider6) PlanAction(_ context.Context, req *tfplugin6.PlanAction_Request) (*tfplugin6.PlanAction_Response, error) {
+	resp := &tfplugin6.PlanAction_Response{}
+
+	actionSchema, ok := p.schema.Actions[req.ActionType]
+	if !ok {
+		return nil, fmt.Errorf("action schema not found for action %q", req.ActionType)
+	}
+
+	ty := actionSchema.ConfigSchema.ImpliedType()
+	configVal, err := decodeDynamicValue6(req.Config, ty)
+	if err != nil {
+		resp.Diagnostics = convert.AppendProtoDiag(resp.Diagnostics, err)
+		return resp, nil
+	}
+
+	linkedResouceSchemas := actionSchema.LinkedResources()
+	inputLinkedResources := make([]providers.LinkedResourcePlanData, 0, len(req.LinkedResources))
+
+	for i, lr := range req.LinkedResources {
```

Comment:
Can you add some comments (here or in top-level method comments) that explain why we aren't sending the planned_state to the provider, when we do for the protocol v5 version? I remember seeing you and James chat about it and I think it would help if that was captured somewhere here